Library/Homebrew/unpack_strategy/dmg.rb

```diff
@@ -87,10 +87,31 @@ def eject(verbose: false)
         return unless path.exist?
 
         if tries > 1
-          system_command! "diskutil",
-                          args:         ["eject", path],
-                          print_stderr: false,
-                          verbose:      verbose
+          disk_info = system_command!(
+            "diskutil",
+            args:         ["info", "-plist", path],
+            print_stderr: false,
+            verbose:      verbose,
+          )
+
+          # For HFS, just use <mount-path>
+          # For APFS, find the <physical-store> corresponding to <mount-path>
+          eject_paths = if disk_info.success?
+            disk_info.plist
+                     .fetch("APFSPhysicalStores", [])
+                     .map { |store| store["APFSPhysicalStore"] }
+                     .compact
+                     .presence || [path]
+          else
+            [path]
+          end
```

Comment:
```suggestion
          eject_paths = disk_info.plist
                     .fetch("APFSPhysicalStores", [])
                     .map { |store| store["APFSPhysicalStore"] }
                     .compact
                     .presence || [path]
```

`success?` is already guaranteed at this point.

Reply:
Thanks. Will amend previous commit with changes once I finish running tests again.

It will also help deal with code coverage as it was probably impossible to reach `else` block.